Feature/gocart suv

lholland-noaa opened the pull request:
Adding interface w/ GOCART Sulfur Volcanic Emissions

lholland-noaa committed:
Initial file (placeholders) for SU Volcanic Emissions

lholland-noaa committed:
Adding datetime utility

lholland-noaa committed:
modifications to comments

lholland-noaa committed:
Initial files to begin SU Volcanic Emissions

lholland-noaa committed:
Reorg of prepmetvars, initial reader placeholder

lholland-noaa committed:
Added example volcanic file and updated code

lholland-noaa committed:
Volcanic updates

lholland-noaa committed:
more updates

lholland-noaa committed:
Updating emission reader

lholland-noaa committed:
Updating module

lholland-noaa committed:
Merge branch 'feature/gocart_suv' of https://github.com/lholland-noaa/CATChem into feature/gocart_suv

lholland-noaa committed:
added notes in comments

lholland-noaa committed:
Updating cmake file to incl SUVolcanic

lholland-noaa committed:
compiles on hera, committing

zmoon commented on src/process/SUVolcanicEmissions/SUVolcanicEmissions_GOCART.F90:
Should this file be removed?

lholland-noaa commented:
Yes!!  Thanks!

On Fri, Jan 17, 2025 at 1:11 PM Zachary Moon ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> On src/process/SUVolcanicEmissions/SUVolcanicEmissions_GOCART.F90
> <https://github.com/ufs-community/CATChem/pull/113#discussion_r1920573460>
> :
>
> Should this file be removed?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/ufs-community/CATChem/pull/113#pullrequestreview-2559671091>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BHNBJUZIDDKE5ECUFWHMRTT2LFBWDAVCNFSM6AAAAABVMJZB36VHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDKNJZGY3TCMBZGE>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Lacey Holland, Ph.D.
Research Physical Scientist
Atmospheric Sciences Modeling Division (ASMD)
NOAA/OAR/ARL
NCWCP, Rm 4257
5830 University Research Court
College Park, Maryland  20740
‪(301) 427-7880‬

lholland-noaa committed:
removing extraneous file

zmoon committed:
Merge branch 'develop' into feature/gocart_suv

zmoon commented on src/core/datetime.F90:
This file seems to be unused, should it be removed?

zmoon replied:
Done in c7f3626

lholland-noaa commented:
Yes!
Also, I made some required updates in the api directory.  There are some
files that needed modified that I missed earlier!

On Wed, Jan 22, 2025 at 5:49 PM Zachary Moon ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> On src/core/datetime.F90
> <https://github.com/ufs-community/CATChem/pull/113#discussion_r1926078681>
> :
>
> This file seems to be unused, should it be removed?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/ufs-community/CATChem/pull/113#pullrequestreview-2568452575>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BHNBJUZNEKEY5VSTXPSTVUT2MAN5VAVCNFSM6AAAAABVMJZB36VHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDKNRYGQ2TENJXGU>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Lacey Holland, Ph.D.
Research Physical Scientist
Atmospheric Sciences Modeling Division (ASMD)
NOAA/OAR/ARL
NCWCP, Rm 4257
5830 University Research Court
College Park, Maryland  20740
‪(301) 427-7880‬

zmoon committed:
fmt

the rc file data lines had trailing single spaces
and it did not have an eof newline

zmoon committed:
Remove unused module

Co-authored-by: lholland-noaa <lacey.holland@noaa.gov>

zmoon commented:
I resolved the merge conflict and got the linter passing here. If you have changes you haven't pushed, you'll have to merge this (the remote branch) into your local branch before you can push.

zmoon commented on src/core/config_mod.F90:
```diff
@@ -1373,4 +1384,78 @@ SUBROUTINE Config_Process_DryDep( ConfigInput, Config, RC )
+         CALL CC_Error( errMsg, RC, thisLoc )
+         RETURN
+      ENDIF
+      Config%drydep_activate = v_bool
```
Should be SUV instead?

lholland-noaa commented:
Yes, it should.  Thank, you!!

On Thu, Jan 23, 2025 at 10:53 AM Zachary Moon ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In src/core/config_mod.F90
> <https://github.com/ufs-community/CATChem/pull/113#discussion_r1927217540>
> :
>
> > +
> +      ! Initialize
> +      RC      = CC_SUCCESS
> +      thisLoc = ' -> at Config_Process_SUVolcanicEmissions (in CATChem/src/core/config_mod.F90)'
> +      errMsg = ''
> +
> +      ! TODO #105 Fix reading of config file
> +      key   = "process%SUVolcanicEmissions%activate"
> +      v_bool = MISSING_BOOL
> +      CALL QFYAML_Add_Get( ConfigInput, TRIM( key ), v_bool, "", RC )
> +      IF ( RC /= CC_SUCCESS ) THEN
> +         errMsg = 'Error parsing ' // TRIM( key ) // '!'
> +         CALL CC_Error( errMsg, RC, thisLoc )
> +         RETURN
> +      ENDIF
> +      Config%drydep_activate = v_bool
>
> Should be SUV instead?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/ufs-community/CATChem/pull/113#pullrequestreview-2570280003>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BHNBJU7THX7SX2FWG7NP7632MEF73AVCNFSM6AAAAABVMJZB36VHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDKNZQGI4DAMBQGM>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Lacey Holland, Ph.D.
Research Physical Scientist
Atmospheric Sciences Modeling Division (ASMD)
NOAA/OAR/ARL
NCWCP, Rm 4257
5830 University Research Court
College Park, Maryland  20740
‪(301) 427-7880‬

lholland-noaa committed:
correcting error in activate flag

zmoon commented on tests/test_suvolcanic.F90:
```diff
@@ -0,0 +1,139 @@
+   SUVolcanicEmissionsState%Activate = .true.
+
+   ! Meteorological State
+   allocate(MetState%BXHEIGHT(MetState%NLEVS))
+   allocate(MetState%DELP(MetState%NLEVS))
+   MetState%NLEVS = 2
```
Looks like this would get the run succeeding
```suggestion
   MetState%NLEVS = 2
   allocate(MetState%BXHEIGHT(MetState%NLEVS))
   allocate(MetState%DELP(MetState%NLEVS))
```

lholland-noaa commented:
Awesome, thanks!

On Fri, Jan 24, 2025 at 10:21 AM Zachary Moon ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In tests/test_suvolcanic.F90
> <https://github.com/ufs-community/CATChem/pull/113#discussion_r1928855918>
> :
>
> > +   allocate(MetState%BXHEIGHT(MetState%NLEVS))
> +   allocate(MetState%DELP(MetState%NLEVS))
> +   MetState%NLEVS = 2
>
> Looks like this would get the run succeeding
> ⬇️ Suggested change
>
> -   allocate(MetState%BXHEIGHT(MetState%NLEVS))
> -   allocate(MetState%DELP(MetState%NLEVS))
> -   MetState%NLEVS = 2
> +   MetState%NLEVS = 2
> +   allocate(MetState%BXHEIGHT(MetState%NLEVS))
> +   allocate(MetState%DELP(MetState%NLEVS))
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/ufs-community/CATChem/pull/113#pullrequestreview-2572943952>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BHNBJUZMQFWN5IAYPRE5OOL2MJK7TAVCNFSM6AAAAABVMJZB36VHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDKNZSHE2DGOJVGI>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Lacey Holland, Ph.D.
Research Physical Scientist
Atmospheric Sciences Modeling Division (ASMD)
NOAA/OAR/ARL
NCWCP, Rm 4257
5830 University Research Court
College Park, Maryland  20740
‪(301) 427-7880‬

zmoon committed:
Set NLEVS before using

zmoon commented on tests/test_suvolcanic.F90:
```diff
@@ -0,0 +1,139 @@
+program test_suvolcanic
+   use CATChem, fp => cc_rk
+   use testing_mod, only: assert
```
Could you update the test to assert some things about the results?

lholland-noaa commented:
Great catches. Thanks!

On Friday, February 7, 2025, Zachary Moon ***@***.***> wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In tests/test_suvolcanic.F90
> <https://github.com/ufs-community/CATChem/pull/113#discussion_r1947288767>
> :
>
> > @@ -0,0 +1,139 @@
> +program test_suvolcanic
> +   use CATChem, fp => cc_rk
> +   use testing_mod, only: assert
>
> Could you update the test to assert some things about the results?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/ufs-community/CATChem/pull/113#pullrequestreview-2603006536>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BHNBJUYL7RILCHYXZ2STAWD2OUZUNAVCNFSM6AAAAABVMJZB36VHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDMMBTGAYDMNJTGY>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Lacey Holland, Ph.D.
Research Physical Scientist
Atmospheric Sciences Modeling Division (ASMD)
NOAA/OAR/ARL
NCWCP, Rm 4257
5830 University Research Court
College Park, Maryland  20740
‪(301) 427-7880‬

lholland-noaa committed:
Update test_suvolcanic.F90

Added "assert" command

lholland-noaa committed:
Update test_suvolcanic.F90

Note on data that needs to go into SUVolcanicEmissionsState

zmoon commented:
Closing in favor of #117
